py-polars/polars/datatypes/classes.py

```diff
@@ -538,16 +538,30 @@ class Enum(DataType):
 
     categories: list[str]
 
-    def __init__(self, categories: list[str]):
+    def __init__(self, categories: Iterable[str]):
         """
         A fixed set categorical encoding of a set of strings.
 
         Parameters
         ----------
         categories
-            Categories in the dataset.
+            Valid categories in the dataset.
 
         """
+        from polars.utils.various import ordered_unique
+
+        categories = list(categories)
```

Comment:
I wonder about this type of pattern: if we're converting into a list anyway, isn't it better to leave it up to the user whether he wants to pay the cost of copying the list? Now we're always copying the list which can get expensive for larger lists. I tested with a list of 1 million 3-character strings and calling `list()` on it takes about `1ms`.

So I think it's better to just accept a `list` here instead of an Iterable, unless we're actually iterating over it.

Reply:
Hmm; yes, materialisation should only really occur if given a generator (otherwise we'd consume it in the checks). Will tweak accordingly.